sql-map-jquery-test/src/main/java/com/hazelcast/jet/tests/jquery/tests/AbstractJsonMapTest.java

```diff
@@ -0,0 +1,174 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.jquery.tests;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.jet.sql.impl.connector.map.IMapSqlConnector;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.jet.tests.common.Util;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlService;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_KEY_FORMAT;
+import static com.hazelcast.jet.sql.impl.connector.SqlConnector.OPTION_VALUE_FORMAT;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public abstract class AbstractJsonMapTest extends AbstractSoakTest {
+    protected static final int DEFAULT_QUERY_TIMEOUT_MILLIS = 100;
+    private static final int PROGRESS_PRINT_QUERIES_INTERVAL = 500;
+    protected final String mapName;
+    protected final String sqlQuery;
+    protected final Boolean requiredSort;
+    protected final String expectedJqueryResultString;
+    protected HazelcastInstance client;
+    protected int queryTimeout = propertyInt("queryTimeout", DEFAULT_QUERY_TIMEOUT_MILLIS);
+    protected String jsonInputString;
+    private long begin;
+    private long currentQueryCount;
+    private long lastQueryCount;
+    private long lastProgressPrintCount;
+
+    public AbstractJsonMapTest(String mapName, String inputJsonFile, String sqlQuery, Boolean requiredSort,
+                               String expectedJsonPath, Boolean resultIsArray) throws IOException, URISyntaxException {
+        this.mapName = mapName;
+        this.jsonInputString = readJsonFromFile(inputJsonFile);
+        this.sqlQuery = sqlQuery;
+        this.requiredSort = requiredSort;
+        this.expectedJqueryResultString = retrieveExpectedJsonStructure(jsonInputString, expectedJsonPath,
+                resultIsArray);
+    }
+
+    protected abstract String retrieveExpectedJsonStructure(String expectedJsonInputString, String expectedJsonPath,
+                                                            Boolean resultIsArray);
+
+    protected void runTest() {
+        begin = System.currentTimeMillis();
+
+        SqlService sql = client.getSql();
+
+        String sqlString = "CREATE OR REPLACE MAPPING " + mapName +
```

Comment:
Can we simplified this please? I mean like `"'='" + "json" + "'\n" + ")";` -> `"'='json'\n)";` etc.

Reply:
ok, simplified